ios/brave-ios/Sources/BraveWallet/Crypto/Stores/NFTStore.swift

```diff
@@ -726,28 +725,14 @@ public class NFTStore: ObservableObject, WalletObserverStore {
     isSpam: Bool,
     isDeletedByUser: Bool
   ) {
-    assetManager.updateUserAsset(
-      for: token,
-      visible: visible,
-      isSpam: isSpam,
-      isDeletedByUser: isDeletedByUser
-    ) { [weak self] in
-      guard let self else { return }
-      let selectedAccounts = self.filters.accounts.filter(\.isSelected).map(\.model)
-      let selectedNetworks = self.filters.networks.filter(\.isSelected).map(\.model)
-
-      let unionedSpamNFTs = computeSpamNFTs(
-        selectedNetworks: selectedNetworks,
-        selectedAccounts: selectedAccounts,
-        simpleHashSpamNFTs: simpleHashSpamNFTs
-      )
-
-      (userNFTGroups, _) = buildNFTGroupModels(
-        groupBy: filters.groupBy,
-        spams: unionedSpamNFTs,
-        selectedAccounts: selectedAccounts,
-        selectedNetworks: selectedNetworks
+    Task { @MainActor in
+      await assetManager.updateUserAsset(
+        for: token,
+        visible: visible,
+        isSpam: isSpam,
+        isDeletedByUser: isDeletedByUser
       )
```

Comment:
Why did we switch this up?

Reply:
since this func `updateNFTStatus` is also getting called in `NFTDetailsView` which has a closure `onNFTStatusUpdated`. this closure will get called once `updateNFTStatus` finishes and then calls `update()`. I think it's a good opportunity to consolidate them together by calling `update()` inside `updateNFTStatus` because
1. since now we have balance/metadata caching, calling `update()` is not costy.
2. we can remove the `onNFTStatusUpdated` to make code more clear.